x-pack/plugins/uptime/server/lib/saved_objects/saved_objects.ts

```diff
@@ -34,6 +35,20 @@ export const registerUptimeSavedObjects = (
       type: syntheticsServiceApiKey.name,
       attributesToEncrypt: new Set(['apiKey']),
     });
+
+    encryptedSavedObjects.registerType({
+      type: syntheticsMonitor.name,
+      attributesToEncrypt: new Set([
+        'secrets',
+        /* adding secretKeys to the list of attributes to encrypt ensures
+         * that secrets are never stored on the resulting saved object,
+         * even in the presence of developer error.
+         *
+         * In practice, all secrets should be stored as a single JSON
+         * payload on the `secrets` key. This ensures performant decryption. */
+        ...secretKeys,
```

Comment:
With type safety and tests, this may not necessary, but I wanted to handle our customer's secrets with care. Adding the extra keys does not significantly reduce performance when they're not defined, and could provide us safety from accidental exposing secrets if they are ever accidentally defined.